hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/ClientConnectionManagerImpl.java

```diff
@@ -131,7 +131,8 @@ public ClientConnectionManagerImpl(HazelcastClientInstanceImpl client, AddressTr
         ClientNetworkConfig networkConfig = config.getNetworkConfig();
 
         final int connTimeout = networkConfig.getConnectionTimeout();
-        this.connectionTimeout = connTimeout == 0 ? Integer.MAX_VALUE : connTimeout;
+        final int attemptPeriod = networkConfig.getConnectionAttemptPeriod();
+        this.connectionTimeout = Math.min(connTimeout, attemptPeriod);
```

Comment:
Previously, if `connTimeout == 0` then `this.connectionTimeout` would become `Integer.MAX_VALUE`, whereas now, assuming a larger `attemptPeriod` (for example `3000` which is the default), `this.connectionTimeout` will be set to 0. Doesn't this affect behaviour of code that uses `ClientConnectionManagerImpl.connectionTimeout`?